### PR TITLE
Solved errors with incomplete formulas

### PR DESCRIPTION
**Pull request by LimaBD**

### CODE REVIEW

File: cases.ts [hunk 1]
```diff
@@ -111,7 +111,7 @@ export const rule4Tree: ASTNode = {
           name: 'EQ',
           args: [
             { type: 'path', path: './companyType' },
-            { type: 'value', value: 'nffe' },
+            { type: 'value', value: "'nffe'" },
```
Comment (cwayfinder) [inline cases.ts]: Probably it's better to introduce the `dataType` property here with possible values 'boolean', 'string', 'number', 'array'. What do you think?

File: cases.ts [hunk 2]
```diff
@@ -163,7 +163,7 @@ export const rule5Tree: ASTNode = {
           args: [
             {
               type: 'value',
-              value: ['investment_company', 'custodial_institution', 'depositary_institution', 'specified_insurance_company'],
+              value: "['investment_company', 'custodial_institution', 'depositary_institution', 'specified_insurance_company']",
```
Comment (cwayfinder) [inline cases.ts]: With the new property `dataType` we could preserve the array.

**Comment by cwayfinder** [comment]
I found a related issue. 
In formula `=EQ(../subscription/price, 1000)`, 1000 should be treated as `value`. Currently, it's `variable`.

File: cases.ts (hunk 1 above)
Comment (LimaBD) [reply cases.ts]: I changed this to preserve the quotes in strings and arrays entered by the user, for example if is entered 'nffe', "nffe" or "nffe (incomplete, without last quote)

File: cases.ts (hunk 2 above)
Comment (LimaBD) [reply cases.ts]: to preserve the arrays, i have undo the changes and implemented autocomplete of quotes

**Pull request merged**